# Find the mean of old scheme and new scheme column. (5 points)


In [7]:
import numpy as np
import pandas as pd

In [10]:
titan_sales_df=pd.read_excel(io="Titan_Data.xls")
titan_sales_df['Old Scheme (in thousands)'].mean()
titan_sales_df['New Scheme (in thousands)'].mean()
print("Mean of Old Scheme is {}".format(titan_sales_df['Old Scheme (in thousands)'].mean()))
print("Mean of New Scheme is {}".format(titan_sales_df['New Scheme (in thousands)'].mean()))




Mean of Old Scheme is 68.03333333333333
Mean of New Scheme is 72.03333333333333


# Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)


In [11]:
from scipy.stats import ttest_rel
ttest_rel(titan_sales_df['Old Scheme (in thousands)'],titan_sales_df['New Scheme (in thousands)'])


Ttest_relResult(statistic=-1.5559143823544377, pvalue=0.13057553961337662)

# What conclusion does the test (p-value) lead to? (2.5 points)

#Null Hypothesis : There is no significant Increase in Output Sales. i.e there is no differance between Mean before New Sales Policy and Mean After Sales policy.  ( Population Mean Before Sales Policy = Population Mean after Sales Policy)

#Alternate Hypothesis : New Scheme has significantly Raised Output Sales. i.e Population Mean After Sales Policy > Population Mean Before Sales Polic1y ( Only Upper Tail). Lower Tail will be the Population Mean After Sales Policy < Population Mean Before Sales Policy.

#Since t(static) is not less than Critical T( alpha=.05/2) ( -1.5559 not less than -2.0452) or t(static) is not greater than Critical T( alpha=.05)/2) ( -1.5559 not greater than 2.0452). So We accept Null Hypothesis.

# So New Scheme has not increased the Sales Output

In [30]:
import scipy.stats as stas
help(stas.ttest_1samp)

Help on function ttest_1samp in module scipy.stats.stats:

ttest_1samp(a, popmean, axis=0, nan_policy='propagate')
    Calculate the T-test for the mean of ONE group of scores.
    
    This is a two-sided test for the null hypothesis that the expected value
    (mean) of a sample of independent observations `a` is equal to the given
    population mean, `popmean`.
    
    Parameters
    ----------
    a : array_like
        sample observation
    popmean : float or array_like
        expected value in null hypothesis. If array_like, then it must have the
        same shape as `a` excluding the axis dimension
    axis : int or None, optional
        Axis along which to compute test. If None, compute over the whole
        array `a`.
    nan_policy : {'propagate', 'raise', 'omit'}, optional
        Defines how to handle when input contains nan. 'propagate' returns nan,
        'raise' throws an error, 'omit' performs the calculations ignoring nan
        values. Default is 'propagate'.

# Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:


#        a) The probability of a type 1 error? (2.5 points)



In [54]:
#Null Hypothesis: The New Scheme Mean is equal 73.034 ( 68.034+5)
#Alternate Hypothesis: The New Scheme Mean is Not Equal to  73.034 ( 68.034+5) ( 2 Tailed)
#Alternate Hypothesis: The New Scheme Mean is Greater than  73.034 ( 68.034+5) ( Upper Tailed)
#Alternate Hypothesis: The New Scheme Mean is Less than  73.034 ( 68.034+5) ( Lower Tailed)
# Caluclating the test for  Population Mean  using Single T Test.

import scipy.stats as stas
t_stat,p_value=stas.ttest_1samp(a=titan_sales_df['New Scheme (in thousands)'],popmean=73.034)
print("T_static Value is {} and Probablilty is {}".format(t_stat,p_value))

T_static Value is -0.2277777033787061 and Probablilty is 0.8214173994071824


In [55]:
# So if the average output must increase by £5000 in the scheme compared to the old scheme. 
#( i.e New Scheme Mean must be greater than or equal to 73.034). So 

type1_error= p_value + (1 - p_value)/2

print("The Type 1 Error for the New Scheme Mean >= 73.034 is {}".format(type1_error))

The Type 1 Error for the New Scheme Mean >= 73.034 is 0.9107086997035911


# b) What is the p- value of the hypothesis test if we test for a difference of 5000? (10 points)



In [56]:
# So if the average output must increase by £5000 in the scheme compared to the old scheme. 
#( i.e New Scheme Mean must be greater than or equal to 73.034). So 

print("The probability for the New Scheme Mean >= 73.034 is {}".format(type1_error))

The probability for the New Scheme Mean >= 73.034 is 0.9107086997035911


#        c) Power of the test (5 points)



In [57]:
# Probablilty of Null Hypothesis is False is 
1-p_value
print("The probability for the Null Hypothesis being False. i.e Beta is {}".format(1-p_value))

The probability for the Null Hypothesis being False. i.e Beta is 0.1785826005928176


In [58]:
print("The Power of Test is {}".format(1-(1-p_value)))

The Power of Test is 0.8214173994071824
